metal/v1/model_device.go

```diff
@@ -1,1680 +1,71 @@
+	Facility            *Facility             `json:"facility,omitempty"`
+	HardwareReservation *HardwareReservation  `json:"hardware_reservation,omitempty"`
+	Hostname            string                `json:"hostname,omitempty"`
+	Href                string                `json:"href,omitempty"`
+	Id                  string                `json:"id,omitempty"`
+	ImageUrl            string                `json:"image_url,omitempty"`
+	IpAddresses         []IpAssignment        `json:"ip_addresses,omitempty"`
+	IpxeScriptUrl       string                `json:"ipxe_script_url,omitempty"`
+	Iqn                 string                `json:"iqn,omitempty"`
 	// Prevents accidental deletion of this resource when set to true.
-	Locked *bool        `json:"locked,omitempty"`
-	Metro  *DeviceMetro `json:"metro,omitempty"`
+	Locked bool              `json:"locked,omitempty"`
+	Metro  *AllOfDeviceMetro `json:"metro,omitempty"`
 	// Whether network mode changes such as converting to/from Layer2 or Layer3 networking, bonding or disbonding network interfaces are permitted for the device.
-	NetworkFrozen *bool `json:"network_frozen,omitempty"`
+	NetworkFrozen bool `json:"network_frozen,omitempty"`
 	// By default, servers at Equinix Metal are configured in a “bonded” mode using LACP (Link Aggregation Control Protocol). Each 2-NIC server is configured with a single bond (namely bond0) with both interfaces eth0 and eth1 as members of the bond in a default Layer 3 mode. Some device plans may have a different number of ports and bonds available.
 	NetworkPorts    []Port           `json:"network_ports,omitempty"`
 	OperatingSystem *OperatingSystem `json:"operating_system,omitempty"`
 	// Actions supported by the device instance.
-	Actions            []DeviceActionsInner `json:"actions,omitempty"`
-	Plan               *Plan                `json:"plan,omitempty"`
-	Project            *Project             `json:"project,omitempty"`
-	ProjectLite        *DeviceProjectLite   `json:"project_lite,omitempty"`
-	ProvisioningEvents []Event              `json:"provisioning_events,omitempty"`
+	Actions            []DeviceActions         `json:"actions,omitempty"`
+	Plan               *Plan                   `json:"plan,omitempty"`
+	Project            *Project                `json:"project,omitempty"`
+	ProjectLite        *AllOfDeviceProjectLite `json:"project_lite,omitempty"`
+	ProvisioningEvents []Event                 `json:"provisioning_events,omitempty"`
 	// Only visible while device provisioning
-	ProvisioningPercentage *float32 `json:"provisioning_percentage,omitempty"`
+	ProvisioningPercentage float32 `json:"provisioning_percentage,omitempty"`
 	// Root password is automatically generated when server is provisioned and it is removed after 24 hours
-	RootPassword *string `json:"root_password,omitempty"`
-	ShortId      *string `json:"short_id,omitempty"`
+	RootPassword string `json:"root_password,omitempty"`
```

Comment:
swagger-codegen doesn't use pointers for optional fields of built-in types; that lines up with how packngo does things